release of merlin 5.3-402 and 4.18-414 (#2893)
ocaml/ocaml.org: data/changelog/merlin/2024-23-12-merlin-5.3.502-and-4.18.414.md

---

> title: Merlin 5.3-502 for OCaml 5.2 and 4.18-414 for OCaml 5.1 and 4.14

Reviewer: ```suggestion
title: Merlin 5.3-502 for OCaml 5.2 and 4.18-414 for OCaml 4.14
```

tags: [merlin, platform]
changelog: |
    + merlin binary
      
        - Respect the EXCLUDE_QUERY_DIR configuration directive when looking for cmt files [#1854](https://github.com/ocaml/merlin/pull/1854)
        - Fix exception in polarity search ([#1858](https://github.com/ocaml/merlin/pull/1858) fixes [#1113](https://github.com/ocaml/merlin/issues/1113))
        - Fix type-enclosing results instability. This reverts some overly aggressive deduplication that should be done on the client side. [#1864](https://github.com/ocaml/merlin/pull/1864)

>         + only in 5.3-502
>          - Fix occurrences bug in which relative paths in index files are resolved against the PWD rather than the SOURCE_ROOT [#1855](https://github.com/ocaml/merlin/pull/1855)
>          - Fix `jump to fun` targets not working ([#1863](https://github.com/ocaml/merlin/pull/1863), fixes [#1862](https://github.com/ocaml/merlin/issues/1862))
>          - Fix occurrences not working when the definition comes from a hidden source file [#1865](https://github.com/ocaml/merlin/pull/1865)

Reviewer: The indentation looks wrong here.
```suggestion
    + merlin binary (only in 5.3-502)
        - Fix occurrences bug in which relative paths in index files are resolved against the PWD rather than the SOURCE_ROOT [#1855](https://github.com/ocaml/merlin/pull/1855)
        - Fix `jump to fun` targets not working ([#1863](https://github.com/ocaml/merlin/pull/1863), fixes [#1862](https://github.com/ocaml/merlin/issues/1862))
        - Fix occurrences not working when the definition comes from a hidden source file [#1865](https://github.com/ocaml/merlin/pull/1865)
```

---

> This release introduces **Merlin 5.3-502**, compatible with OCaml 5.2, and **4.18-414**, compatible with OCaml 5.1 and 4.14. Key updates include `EXCLUDE_QUERY_DIR` for better file management, resolving exceptions in polarity search, 

Reviewer: ```suggestion
This release introduces **Merlin 5.3-502**, compatible with OCaml 5.2, and **4.18-414**, compatible with OCaml 4.14. Key updates include `EXCLUDE_QUERY_DIR` for better file management, resolving exceptions in polarity search, 

```

> and stabilizing type-enclosing results. For users of 5.3-502, additional enhancements address path resolution issues, improve `jump to fun` functionality, and ensure proper handling of occurrences for hidden source files.

Reviewer: ```suggestion
and stabilizing type-enclosing results. For users of 5.3-502, additional enhancements address path resolution issues, fixes `jump to fun` functionality, and ensure proper handling of occurrences from hidden source files.
```
